pkg/asset/agent/workflow/commons.go

```diff
@@ -0,0 +1,12 @@
+package workflow
+
+// AgentWorkflowType defines the supported
+// agent workflows.
+type AgentWorkflowType string
+
+const (
+	AgentWorkflowTypeInstall  AgentWorkflowType = "install"
+	AgentWorkflowTypeAddNodes AgentWorkflowType = "addnodes"
```

Comment:
> I'm not sure I understand the `AgentWorkflow` asset. Are the different workflows enabled by which file is loaded?

The AgentWorkflow reads the current current workflow from the `.agentworkflow`, so that it could be used by other assets to determine their own logic (see a sample of its usage in the subsequent PR #8009).
The file remains always the same, it's just its content that is different based on the current workflow enabled.
When fetched for the first time, if no existing `.agentworkflow` file was found, then the `AgentWorkflow` falls back to the default and generates the `install` worklow - this ensures the backward compatibility for the agent assets behaviors for the standard command in the `openshift-install` binary, the `agent create image` one.
For the `node-joiner` instead, a different workflow is preliminarly activated by the [add nodes command](https://github.com/openshift/installer/pull/7997/files#diff-6655d118f522d77bf3f4a0a1ef3cc895a7aa987b7789c11296d958329e069cdcR24).

Reply:
(as an additional note, not related to this epic, as I'm progressing in #8009, it seems to me that this mechanism could also be used in future to represent and make more explicit/clear another existing agent workflow, the "install with ZTP manifests" one: right now it's implicitly implemented by the checking everytime the availability of the install-config asset in every agent asset where it's needed, as for example [here](https://github.com/openshift/installer/blob/0481a931116619c82782f5a0ff069404dc6817a7/pkg/asset/agent/manifests/clusterdeployment.go#L48))

Reply:
Thanks for clarifying :)